mantidimaging/core/data/images.py

```diff
@@ -66,6 +66,20 @@ def metadata_save(self, f):
     def metadata_saves(self):
         return json.dumps(self.properties)
 
+    def record_parameters_in_metadata(self, func, *args, **kwargs):
+        if 'operation_history' not in self.properties:
+            self.properties['operation_history'] = []
+
+        def accepted_type(o):
+            return type(o) in [str, int, float, bool, tuple]
```

Comment:
Is there a reason we accept `tuple` but not `list` or `ndarray`?

Reply:
The ones listed are the ones that are definitely used (apart from `str`) that will not result in trying to save entire image stacks to a JSON file (which would happen if we accepted `ndarray`).